### ISSUE TITLE
Doxy comment nearest entity command should favor following function over preceeding

### ISSUE THREAD

**Issue by JohannesMP** [20Tauri/DoxyDoxygen#15]
When the cursor is placed between two un-documented functions, such as this:

```
Bar::Bar(int derp) : derp_(derp)
{}

int Bar::GetDerp()
{
  return derp_;
}
```

The comment nearest entity command (`super`+`alt`+`Q` on mac) will unintuitively favor the preceding function:

![doxygen2](https://cloud.githubusercontent.com/assets/662874/10037824/73067c60-6170-11e5-8a58-8eb010d018a3.gif)

**Comment by 20Tauri** [comment]
Currently, 3 lines up have the same weight than 1 line down.
In my workflow. I write the function name and then I press <kbd>Alt</kbd>+<kbd>Q</kbd>
It's can be a parameter, or just a lake in documentation...

**Comment by JohannesMP** [comment]
It would be great if I could change the weighting with parameters

**Comment by 20Tauri** [comment]
In next release, I will introduce this parameter (name is hard to define)

```
    // "Comment nearest entity command" uses this value to locate the "nearest" function.
    // If the value is 3, the plugin prefers go up to 2*3-1=5 lines up, instead of
    // 2 lines down.
    "distance_for_bottom_line_over_top_line": 3,
```

In current version (without this parameter), the value is 3.
Why ? Because, I usually write the prototype before pressing <kbd>ALT</kbd>+<kbd>Q</kbd>.
You use C++, there's some advanced functionality that parse the function body when parameter type is not part of the language (ex: javascript).

I will probably keep this value... Even if for a C++ programmer, with a looooonnnnng template this may be "surprising"...

If someone want to join the conversation, user experience is welcome.

**Comment by 20Tauri** [comment]
Released in 0.20.0
